components/scream/src/physics/share/physics_constants.hpp

```diff
@@ -95,6 +95,8 @@ struct Constants
   static constexpr Scalar f2r           = 0.32;
   static constexpr Scalar nmltratio     = 0.2; // ratio of rain number produced to ice number loss from melting
   static constexpr Scalar basetemp      = 300.0;
+  static constexpr Scalar maxlen        = 20000.0;
+  static constexpr Scalar length_fac    = 0.5;
 
```

Comment:
These were constants used in the shoc.F90 `compute_shoc_mix_shoc_length` subroutine (and other subroutines), but not defined in this file. Is it appropriate to add them? Or should I just locally define these values?

Reply:
In the past I've recommended a SHOC-specific constants data structure. I continue to recommend that, and you should put these constants there.

Reply:
Here's an example with verbose constant names that include units: https://github.com/E3SM-Project/haero/blob/master/haero/physical_constants.hpp

It's a to-do item to add `ekat::units` support, as well as to unify the way scream handles such constants.

Reply:
@ambrad @pbosler I added a new file: `src/physics/shoc/physics_constants.hpp` containing these 2 constants (and removed from here). Is this what you were thinking? If so, should I find all the constants currently being used in the various shoc functions and add them here?

Reply:
Can be done in another PR so you can focus just on what's needed for this one.

Reply:
I actually added a file called [`shoc_constants.hpp`](https://github.com/E3SM-Project/scream/blob/ede6bb1d2b6ad14e03cc7bb60b91547bdeea9968/components/scream/src/physics/shoc/shoc_constants.hpp) in my PR #578. If this PR got merged first (which I think it will be), I will add my changes to this file. I still think it is better to name it "shoc_constants.hpp" just to make it more clear that these constants are SHOC specific.

@ambrad : I am of the view that constants which are specific to a function should stay in the function file. It make it easier as we can view the constants right in the file. Constants which are shared among at least two functions, should appear in shoc_constants.hpp. Is there an advantage to put all constants in shoc_constants.hpp file?

Reply:
@singhbalwinder makes sense and fine with me with one detail: If a constant appears at the top of shoc.F90 b/c it's intended to be a tunable parameter, it should be captured as such in a SHOC's constants struct.

Reply:
Thanks and I agree with you on that.